lib/cstruct.ml

```diff
@@ -314,7 +330,10 @@ let iter lenfn pfn buf =
           body := None;
           None
         |Some plen ->
-          let p,rest = split buf plen in
+          incr i;
+          let p,rest =
+            try split buf plen with Invalid_argument _ -> err_iter !i plen
```

Comment:
The info of this error is not expressed in terms of the client provided arguments, or am I missing something ?